### ISSUE TITLE
table cells starting with = creating empty node to the left

### ISSUE THREAD

**Issue by pkra** [zorkow/speech-rule-engine#800]
The typical long equation pattern, `\begin{align} a & = b \\ & = c ... \end{align}` creates an empty node to the left of the equal sign. In speech, this comes out as "blank" in Nemeth it's empty.

I've been wondering why it's there in the first place. It seems to create unnecessary (and potentially confusing) noise for the user.

**Comment by zorkow** [comment]
Is the blank in all rows or only in the second row, i.e. where there is an empty cell.

**Comment by pkra** [comment]
I see it in all the rows. Also for inequalities.

Aside: I've now noticed that the empty first column get something like "1st Column Blank" for speech but all whitespace in Nemeth which is also not ideal.

Here's a fuller example:

```
\begin{align} \mathcal{Y}(M,[g])^2 & = \frac{1}{{\operatorname {Vol}_{\overline{g}}(M)}} \left(\int _{M}\overline{S} dV_{\overline{g}}\right)^2 \\ & = \int _{M}\overline{S}^{2} dV_{\overline{g}}\\ & \geq \int _{M}\left(\overline{S}^{2} -12|\mathring{\overline{Ric}}|^2 \right) dV_{\overline{g}}\\ & = \int _{M}\left(S^{2} -12|\mathring{Ric}|^2 \right) dV_{g}, {\tag{3.11}} \end{align}
```

**Comment by dpvc** [comment]
> In speech, this comes out as "blank" in Nemeth it's empty.

That's probably my fault.  I had made the explorer treat a blank entry as "blank", since it seemed strange to move to something and get no feedback.  I didn't think about the Braille version.

> an empty node to the left of the equal sign.

This is so that the equal sign is an infix operator rather than a prefix one.  In TeX, one would do `{} = x` to get that, and that is what the `align` environment does, even in LaTeX.

**Comment by pkra** [comment]
> That's probably my fault. I had made the explorer treat a blank entry as "blank", since it seemed strange to move to something and get no feedback. I didn't think about the Braille version.

I don't think that's a mistake as such. It's an ARIA and WCAG failure to have an empty aria-label (or aria-braillelabel). Braille users will also get aria-label content if there's no aria-braillelabel. [Edit: I realize I'm making assumptions - I don't know how MathJax's client-side approach works these days.]

As I brought up in https://github.com/Speech-Rule-Engine/speech-rule-engine/issues/736#issuecomment-3028489919 I see a problem in that SRE sometimes intentionally creates empty speech output (instead of no speech attribute). This makes it hard to identify bugs (like #799).

> This is so that the equal sign is an infix operator rather than a prefix one. In TeX, one would do {} = x to get that, and that is what the align environment does, even in LaTeX.

Yes, I'm aware of that. But I think it's a mistake to include this in SRE output. I don't think users benefit but it can cause problems for them.

**Comment by pkra** [comment]
Still seeing this in v5-beta.3.

**Comment by pkra** [comment]
Still seeing this in 5.0.0-rc.4.

**Comment by pkra** [comment]
I've noticed more cases for this behavior. Unfortunately I cannot edit the title but I think the broader question I have is: why expose empty nodes? 

In particular in the context of SRE's skeleton output - client side rendering seems to ignore them sometimes.

E.g. `$^{+}$`. The skeleton output will create an empty base (in mathspeak and nemeth), but e.g. on mathjax.org#demo the explorer will not expose the empty base. 

Similarly in more complex situations like `$\alpha \in \mathbb{Z}_{\ge 0}^{k}$` (to the left of the `\ge`)

IIUC SRE annotates these elements with `data-semantic-type="empty"` but I'm wondering why they are included in the skeleton if they're supposed to be ignored.  It just makes it harder to filter them out.
